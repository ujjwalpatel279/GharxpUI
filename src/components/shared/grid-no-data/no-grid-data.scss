// ========================================================================
// COMPONENTS/SECTIONS/GRID
// ========================================================================

@import '~sassHelpers';

.c-grid-no-data {
  &__message {
    font-weight: normal;
    color: color(grey3);
  }

  img {
    @include bp(20, max) {
      transform: scale(0.7);
    }
  }
}
