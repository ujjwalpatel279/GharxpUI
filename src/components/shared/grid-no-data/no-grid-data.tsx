import React, { FunctionComponent, ReactElement } from 'react';
import NoApplicants from '../../../images/no-applicants.svg';
import NoClients from '../../../images/no-clients.svg';

import './no-grid-data.scss';

interface NoGridDataProps {
  image: 'no-clients' | 'no-applicants';
  message: string;
}

export const NoGridData: FunctionComponent<NoGridDataProps> = ({ image, message }): ReactElement => {
  let imageSrc = NoApplicants;
  switch (image) {
    case 'no-clients':
      imageSrc = NoClients;
  }
  return (
    <div className="c-grid-no-data">
      <img src={imageSrc} className="c-grid-no-data__image" alt="no-data"></img>
      <div className="c-grid-no-data__message">{message} </div>
    </div>
  );
};
