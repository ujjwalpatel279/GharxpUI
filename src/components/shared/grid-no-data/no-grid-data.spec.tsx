import React from 'react';

import { NoGridData } from './no-grid-data';

import { render, cleanup } from '@testing-library/react';

describe('No Grid Data Component', () => {
  afterEach(cleanup);
  it('renders the no grid data component', () => {
    const { container } = render(<NoGridData image="no-clients" message="" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the no grid data message', () => {
    const message = 'there are no clients';
    const { getByText } = render(<NoGridData image="no-clients" message={message} />);
    expect(getByText(message)).toBeInTheDocument();
  });
});
