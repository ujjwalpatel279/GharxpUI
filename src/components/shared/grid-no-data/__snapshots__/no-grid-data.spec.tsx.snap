// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No Grid Data Component renders the no grid data component 1`] = `
<div
  class="c-grid-no-data"
>
  <img
    alt="no-data"
    class="c-grid-no-data__image"
    src="[object Object]"
  />
  <div
    class="c-grid-no-data__message"
  >
    
     
  </div>
</div>
`;
